flang/lib/Lower/IntrinsicCall.cpp

```diff
@@ -281,6 +283,9 @@ static constexpr IntrinsicHandler handlers[]{
     {"mod", &I::genMod},
     {"nint", &I::genNint},
     {"present", &I::genPresent, {{{"a", asInquired}}}, /*isElemental=*/false},
+    {"scan", &I::genScan, {{ {"string", asAddr}, {"set", asAddr}, 
+                          {"back", asAddr}, {"kind", asValue} }}, 
```

Comment:
@jeanPerier : It is a little unclear looking at this table if these argument specifiers are optional or required.

Reply:
I could add a comment. As soon an there is one argument that you do not want by value, you need to add it here, and it is not wrong to add explicit ones per value (like `kind` here) when specifying other argument, though it can be skipped. I guess for simplicity, it should be all or nothing.

- All by value: do not provide argument specifiers.
- At least one not by value: specify all argument specifiers.

Reply:
@jeanPerier If you want, I can incorporate the above into the the f18-llvm-project/blob/fir-dev/flang/docs/tutorials/addingIntrinsics.md document.

Reply:
Thanks Mark, it would be great. I updated the comment above the table in #703 to better described this, but having it in the doc is still a good idea.